### ISSUE TITLE
Unable to load remote states in 2.23.1

### ISSUE THREAD

**Issue by adamlc** [AzBuilder/terrakube#1360]
### Bug description 🐞

When trying to use `terraform_remote_state` resource it no longer works and errors with the following:

```
Error: Error downloading state: 403 Forbidden

Error: data.terraform_remote_state.****: Unable to read remote state
```

Nothing particularly useful in the executor or API logs. I have made sure my team has the new manage state permission. Not sure if it's related!

### Steps to reproduce

Reference a state from another workspace and run a plan

**Comment by alfespa17** [comment]
hello @adamlc quick question are you using the remote or local execution?

**Comment by adamlc** [comment]
Remote :)

**Comment by alfespa17** [comment]
@adamlc this should be fixed in version 2.23.2

**Comment by adamlc** [comment]
@alfespa17 I can confirmed this is fixed. Thank you once again for the quick turnaround, you guys are awesome :D
